graylog2-server/src/main/java/org/graylog2/users/UserImpl.java

```diff
@@ -250,6 +259,22 @@ public boolean isLocalAdmin() {
         return false;
     }
 
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    @Override
+    public Set<String> getRoleIds() {
+        final List<ObjectId> roles = (List<ObjectId>) fields.get(ROLES);
```

Comment:
When using `firstNonNull((List<ObjectId>) fields.get(ROLES), Collections.emptyList())` we could save the null check in the next line(s).